src/redux/tasks.js

```diff
@@ -0,0 +1,41 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const taskSlice=createSlice({
+    name:'tasks',
+    initialState:[],
+    reducers:{
+        ADDTASK:(state,action)=>{
+           state=[...state,action.payload]
+           return state
+        },
+        REMOVETASK:(state,action)=>{
+            state=state.filter((e)=>e.id!==action.payload.id)
+            return state
+        },
+        UPDATETASK:(state,action)=>{
+            const update=state.map((e)=>{
+            if (e.id===action.payload.id) {
+                return {
+                    ...e,
+                    task:action.payload.task,
+                    completed:action.payload.completed,
+                    isEdited:action.payload.isEdited
+                }
+            }else if(action.payload.isEdited===true){
+                return {
+                    ...e,
+                    isEdited:false
+                }
+            }else{
+                return e
+            }
```

Comment:
else statements are not required with return. This makes the code look complex. Use the same early return as described above.